### DISCUSSION TITLE
PRs not squashed on merge

### DISCUSSION THREAD

**Discussion by MSDN-WhiteKnight** [dotnet/roslyn#62272] [General] [answered]
Any reason some PRs are not squashed on merge in roslyn? This produces odd commit history with every small work-in-progress commit included:

![image](https://user-images.githubusercontent.com/35516665/176597746-8990c23b-f920-42d1-9f1e-5bfe4bd255f6.png)

I think other dotnet repositories always use squash merge.

**Comment by davidwengier** [comment]
Personal preference I guess. I always squash my PRs 🤷‍♂️

**Reply by cartermp** [reply]
I would always squash @dsyme's PRs but that's because I wanted to make sure that myself and @KevinRansom could catch up.

**Reply by davidwengier** [reply]
There definitely are certain people who don't squash and certain people who are at the top of the leaderboard of commits, and perhaps those certain people are the same certain people 🤷‍♂️

**Comment by jcouv** [comment, marked as answer]
The compiler team has a guideline to squash PRs that affect the compiler codebase. Some accidentally get through because we don't have enough enforcement for that discipline.

> I think other dotnet repositories always use squash merge.

~~That seems to be the case. I can only see squashes in the runtime repo, but couldn't find any explicit discussion/documentation.~~
I confirmed that the runtime repo has a "squash everything" policy, unless there's a strong reason to make an exception.
